server/game/cards/06-CotE/KikuMatsuri.js

```diff
@@ -0,0 +1,28 @@
+const ProvinceCard = require('../../provincecard.js');
+const AbilityDsl = require('../../abilitydsl');
+
+class KikuMatsuri extends ProvinceCard {
+    setupCardAbilities() {
+        this.action({
+            title: 'Honor a character home from each side',
+            targets: {
+                myCharacter: {
+                    cardCondition: (card, context) => card.isParticipating() && card.controller === context.player,
```

Comment:
For both of these it would be better to have `cardType: CardTypes.Character` and `controller: Players.Self/Opponent`.  You should always have a cardType in a target, because it makes the prompt work correctly, and using controller (rather than adding the condition to the cardCondition) will make the targeting prompt work better

Reply:
No problem, I'll fix that up shortly